core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -817,6 +824,7 @@ public static class Builder {
     private long currentSnapshotId;
     private List<Snapshot> snapshots;
     private final Map<String, SnapshotRef> refs;
+    private final Map<Long, List<StatisticsFile>> statisticsFiles;
```

Comment:
Why use a multimap? Shouldn't `setStatistics` replace the previous file? I think that's the behavior that is currently implemented, so I think this should just be a regular map to ensure there aren't somehow multiple files for a snapshot ID.

Reply:
> Why use a multimap? 

This is so per https://github.com/apache/iceberg/pull/5021#discussion_r936919462

Please clarify where I can use a Map and where I should not, the lifecycle of these metadata objects is still not crystal clear to me

Reply:
Okay, so the idea is to simply preserve cases where there are multiple stats files? I guess I'm okay with that. I'd also be fine with discarding one of them.